### PR TITLE
Added --generate-sources-and-exit option

### PR DESCRIPTION
**Pull request by Artem-B**
When specified, TensileCreateLibray generates asm and source kernels
and exists without compiling them. Assembly sources will be placed in
the output directory.

Also added -v and -j to control output verbosity and the number of
threads launched by the script.

### CODE REVIEW

**Comment by Artem-B** [comment]
I didn't test the patch with the master branch, so I'm not sure if it works there.
Original version of the patch is on rocm-3.7.x branch and can be found here:
https://github.com/Artem-B/Tensile/commit/6893c8dedcbc30f282b04fe23301098e286692bb

File: Tensile/bin/TensileCreateLibrary [hunk 1]
```diff
@@ -21,6 +21,15 @@
 ################################################################################
 # This script only gets called by CMake
 
+import google3
```
Comment (sdquiring) [inline Tensile/bin/TensileCreateLibrary:24]: What is this module?

File: Tensile/KernelWriter.py [hunk 2]
```diff
@@ -3268,18 +3268,23 @@ def getSourceFileString(self, kernel):
...
-        self.writeByteArrayScript()
-
         asmPath = self.getAssemblyDirectory()
-        coFile = self.getSingleCodeObjectFile(kernel)
         kernelName = self.getKernelName(kernel)
 
-        if globalParameters["CodeFromFiles"] or globalParameters["NewClient"] > 1:
-          # I guess in this case we are making sure that the code object file exists by executing the code
-          # above but we aren't placing it into the source.
+        if globalParameters["GenerateSourcesAndExit"]:
```
Comment (sdquiring) [inline Tensile/KernelWriter.py:3274]: A new global parameter should get a default value by assigning it in [Common.py](https://github.com/ROCmSoftwarePlatform/Tensile/blob/develop/Tensile/Common.py#L53).

File: Tensile/TensileCreateLibrary.py [hunk 3]
```diff
@@ -1280,6 +1282,12 @@ def TensileCreateLibrary():
                           default=False, help="Output manifest file with list of expected library objects and exit.")
   argParser.add_argument("--library-format", dest="LibraryFormat", choices=["yaml", "msgpack"], \
       action="store", default="msgpack", help="select which library format to use")
+  argParser.add_argument("--generate-sources-and-exit",   dest="GenerateSourcesAndExit", action="store_true",
+                          default=False, help="Output source files only and exit.")
+  argParser.add_argument("-j",   dest="CpuThreads", type=int,
+                          default=-1, help="Number of parallel jobs to launch.")
+  argParser.add_argument("-v",   dest="PrintLevel", type=int,
+                          default=-1, help="Set printout verbosity level.")
```
Comment (sdquiring) [inline Tensile/TensileCreateLibrary.py:1290]: Good idea to make this more canonical.

**Review by sdquiring — CHANGES_REQUESTED** [review]
Generally looks good.  I switched this to merge into develop since that fits with our development process.

Our CI will not run successfully without the default value and with the `google3` import.

**Review by cgmillette — COMMENTED** [review]
I will sanity test this tomorrow and make sure our regular workflow isn't impacted. I see CI is failing, looks like due to uninitialized globalParameters entry. Should be easy fix.

So far, looks good minus the package import. Looks like something for bazel?

File: Tensile/bin/TensileCreateLibrary (hunk 1 above)
Comment (Artem-B) [reply Tensile/bin/TensileCreateLibrary:24]: This is an unintentional local change needed to run in our environment. I'll remove it.

File: Tensile/KernelWriter.py (hunk 2 above)
Comment (Artem-B) [reply Tensile/KernelWriter.py:3274]: Will do.

File: Tensile/TensileCreateLibrary.py (hunk 3 above)
Comment (Artem-B) [reply Tensile/TensileCreateLibrary.py:1290]: Do you mean to make them `--jobs` and `--verbose`? Or do you prefer them to match `CpuThreads`/`PrintLevel`?

File: Tensile/KernelWriter.py (hunk 2 above)
Comment (Artem-B) [reply Tensile/KernelWriter.py:3361]: Done.

File: Tensile/bin/TensileCreateLibrary (hunk 1 above)
Comment (Artem-B) [reply Tensile/bin/TensileCreateLibrary]: Done.

File: Tensile/TensileCreateLibrary.py (hunk 3 above)
Comment (Artem-B) [reply Tensile/TensileCreateLibrary.py]: Updated to accept both `--jobs|-j` and `--verbose|-v`.

**Review by cgmillette — APPROVED** [review]

**Review by sdquiring — APPROVED** [review]

**Pull request merged**